File: languages/go/entity/key/local_certificate_args.go
package key

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type LocalCertificateParams struct {
	KeyType          KeyType
	Password         string
	Subject          SubjectCertificateParams
	ExpirationMonths int32
}

func NewLocalCertificateParamsFromProto(s *proto.LocalCertificateParams) LocalCertificateParams {
	if s == nil {
		return LocalCertificateParams{}
	}
	return LocalCertificateParams{
		KeyType: KeyType(s.GetKeyType()),
		Subject: SubjectCertificateParams{
			CommonName:       s.GetSubject().GetCommonName(),
			Organization:     s.GetSubject().Organization,
			OrganizationUnit: s.GetSubject().OrganizationalUnit,
			Location:         s.GetSubject().Location,
			State:            s.GetSubject().State,
			Country:          s.GetSubject().Country,
		},
		ExpirationMonths: s.GetExpiration(),
	}
}

func (s LocalCertificateParams) ToProto() *proto.LocalCertificateParams {
	return &proto.LocalCertificateParams{
		KeyType:  KeyTypeToProto[s.KeyType],
		Password: s.Password,
		Subject: &proto.CertificateSubject{
			CommonName:         s.Subject.CommonName,
			OrganizationalUnit: s.Subject.OrganizationUnit,
			Organization:       s.Subject.Organization,
			Location:           s.Subject.Location,
			State:              s.Subject.State,
			Country:            s.Subject.Country,
		},
		Expiration: s.ExpirationMonths,
	}
}

File: languages/js/src/entity/key/local_certificate_args.ts
import * as keysEntitiesProto from "../../bridge/proto/keys_entities";
import { KeyType } from "./key_type";
import { SubjectCertificateParams } from "./subject_certificate_params";

/**
 * Represents the parameters for generating a local certificate.
 */
export class LocalCertificateParams {
  public keyType: KeyType;
  public subject: SubjectCertificateParams;
  public password: string;
  public expiration: number;

  /**
   * Constructs an LocalCertificateParams object with the specified parameters.
   * @param keyType 
   * @param subject 
   * @param password 
   * @param expiration 
   */
  constructor(
    keyType: KeyType,
    subject: SubjectCertificateParams,
    password: string,
    expiration: number
  ) {
    this.keyType = keyType;
    this.subject = subject;
    this.password = password;
    this.expiration = expiration;
  }

  public toProto(): keysEntitiesProto.LocalCertificateParams {
    return keysEntitiesProto.LocalCertificateParams.fromPartial({
      keyType: KeyType.toProto(this.keyType),
      subject: this.subject.toProto(),
      password: this.password,
      expiration: this.expiration
    });
  }
}
